src/core_modules/capture-core/components/ListView/Filters/Contents/FilterSelectorContents.component.js

```diff
@@ -13,107 +12,87 @@ import {
-        // $FlowFixMe[invalid-computed-prop] automated comment
-        [filterTypesObject.ASSIGNEE]: AssigneeFilter,
-    };
+const OptionSetFilterWithButtons = withButtons()(
+    OptionSetFilter,
+);
 
-    static hasStylesHOC = [
-        // $FlowFixMe[prop-missing] automated comment
-        filterTypesObject.NUMBER,
-        // $FlowFixMe[prop-missing] automated comment
-        filterTypesObject.INTEGER,
-        // $FlowFixMe[prop-missing] automated comment
-        filterTypesObject.INTEGER_POSITIVE,
-        // $FlowFixMe[prop-missing] automated comment
-        filterTypesObject.INTEGER_NEGATIVE,
-        // $FlowFixMe[prop-missing] automated comment
-        filterTypesObject.INTEGER_ZERO_OR_POSITIVE,
-        // $FlowFixMe[prop-missing] automated comment
-        filterTypesObject.DATE,
-        // $FlowFixMe[prop-missing] automated comment
-        filterTypesObject.BOOLEAN,
-        // $FlowFixMe[prop-missing] automated comment
-        filterTypesObject.TRUE_ONLY,
-        filterTypesObject.ASSIGNEE,
-    ];
+const selectorContentsForTypes = {
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.TEXT]: TextFilter,
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.NUMBER]: NumericFilter,
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.INTEGER]: NumericFilter,
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.INTEGER_POSITIVE]: NumericFilter,
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.INTEGER_NEGATIVE]: NumericFilter,
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.INTEGER_ZERO_OR_POSITIVE]: NumericFilter,
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.DATE]: DateFilter,
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.BOOLEAN]: BooleanFilter,
+    // $FlowFixMe[prop-missing] automated comment
+    [filterTypesObject.TRUE_ONLY]: TrueOnlyFilter,
+    // $FlowFixMe[invalid-computed-prop] automated comment
+    [filterTypesObject.ASSIGNEE]: AssigneeFilter,
+};
 
-    getContentsComponent() {
-        const { type, optionSet } = this.props;
-        // important to use PureComponent for this
-        if (optionSet && optionSet.options && optionSet.options.length <= MAX_OPTIONS_COUNT_FOR_OPTION_SET_CONTENTS) {
-            return FilterSelectorContents.getOptionSetComponent();
-        }
-        const SelectorContent = FilterSelectorContents.selectorContentsForTypes[type];
-        if (FilterSelectorContents.hasStylesHOC.includes(type)) {
-            return withButtons()(
-                SelectorContent,
-            );
+const useContents = ({ filterValue, classes, type, options, multiValueFilter, ...passOnProps }) => {
+    const [FilterContents, ofTypeOptionSet] = useMemo(() => {
+        if (options && options.length <= MAX_OPTIONS_COUNT_FOR_OPTION_SET_CONTENTS) {
+            return [OptionSetFilterWithButtons, true];
```

Comment:
Perhapsyou dont really refactor this component but since we are touching I think I can point this out.

I am asking myself if `return`ing a component from this hook is a viable option? 

Cant we use the hook to derive which kind of component we want to render and inside the component `FilterSelectorContentsPlain`  decide if we wanna display the
`withButtons()(OptionSetFilter)` or the `[withButtons()(TypeFilter)`? 

I guess some arguments regarding what I am suggesting has advantages would bethe fact that useContents does many things. Also perhaps you can say that hooks arent really meant to return whole components but more calculate stuff.

It feels now that reading the lines 91-93 
```
    return (
        <div className={classes.container}>
            {contents}
        </div>
    );
};
```

I dont get any information about what contents is. While if we had something between these lines 

```
    return (
        <div className={classes.container}>
            {ofTypeOptionSet ? <OptionSetFilterWithButtons /*pass props here */ /> : <FilterWithButtons /*pass props here */ />}
        </div>
    );
};
```

I would see what is goin on and would really care about what is happening inside the hook. I would imagine that are some calculations that return a boolean.

Of course this is perhaps a more of a nitty peaky comment but yeah I feel that changing this thing around would give the reader of a better view of what this component rendering